### ISSUE TITLE
Concat multi page tables

### ISSUE THREAD

**Issue by vinayak-mehta** [camelot-dev/camelot#8]
Would be nice to have a way to merge tables which span multiple pages.

**Comment by udayrddy** [comment]
@vinayak-mehta I would like to contribute to this. However, I would like to know your expectations on the scenarios and how to handle those.

**Comment by c0nb4** [comment]
The way I'm Doing this in my personal project is with

`pd.concat(self.list_of_dfs)`

The only problem I see is when tables have different column-names. So I just Rename them 


`names = self.list_of_dfs[0].columns.tolist()
        for df in self.list_of_dfs:
            df.columns = names`

**Comment by vinayak-mehta** [comment]
@akshowhini This is easy to do when the number and name of columns are the same, which doesn't happen very often. A robust way to do this would be to group multiple tables from different pages based on partially matching the column names (based on some threshold) and concatenate them.

**Comment by vinayak-mehta** [comment]
This issue is low priority as there's no _general_ way to merge tables spanning multiple pages across millions of different types of table structures.

**Comment by bosd** [comment]
Tracking this request alongside #8, #133, #357, #531, and #628 as the same feature: vertically concatenate tables that span multiple pages.

@TimothyOfDelphi's [#628](https://github.com/camelot-dev/camelot/issues/628) is the most useful summary — it includes a working proof-of-concept (`vstack_tables` + `stack_contiguous_tables`) that takes a `TableList` and stacks tables sharing column count.

Implementation direction settled on: a **post-extraction helper** on `TableList`, not a parameter to `read_pdf` or a push-down into the PDF reader. The 2020 triage by Vinayak was right that there's no general algorithm — the column-count rule from #628 covers ~the common case, and stricter matching (column-header similarity, row-count divisibility, etc.) can be layered on as opt-in flags. Working on lifting the POC into a `TableList.stack_contiguous()` method now; once it lands this issue will be closed against that PR. Please 👍 #628 if this matches what you wanted, or comment there if your PDF needs a different stitching rule.
